### ISSUE TITLE
Bug report- Query Parameters in apim operations not working as intended

### ISSUE THREAD

**Issue by jioblack** [aztfmod/terraform-azurerm-caf#1520]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Version of the module you are using

5.6.5

### Rover Version

none

### Terraform Version

1.3.0

### AzureRM Provider Version

=> 2.99

### Affected Resource(s)/Data Source(s)

api_management_api_operation

### Terraform Configuration Files

```hcl
global_settings = {
  default_region = "region1"
  regions = {
    region1 = "australiaeast"
  }
}

resource_groups = {
  rg1 = {
    name   = "example-agw"
    region = "region1"
  }
}


api_management = {
  apim1 = {
    name   = "example-apim"
    region = "region1"
    resource_group = {
      key = "rg1"
    }
    publisher_name  = "My Company"
    publisher_email = "company@terraform.io"

    sku_name = "Developer_1"
  }
}

api_management_api = {
  apimapi1 = {
    name = "example-api"
    resource_group = {
      key = "rg1"
    }
    api_management = {
      key = "apim1"
    }
    revision     = "1"
    display_name = "Example API"
    path         = "example"
    protocols    = ["https"]

    import = {
      content_format = "swagger-link-json"
      content_value  = "http://conferenceapi.azurewebsites.net/?format=json"
    }
  }
}


api_management_api_operation = {
  apimapio1 = {
    operation_id = "sample"
    api = {
      key = "apimapi1"
    }
    api_management = {
      key = "apim1"
    }
    resource_group = {
      key = "rg1"
    }

    display_name = "Get WishLists"
    method       = "GET"
    url_template = "/wishlists"
    description  = "Get WishList"
    request = {
      query_parameter = {
        name     = "test"
        required = false
        type     = "string"
      }
            query_parameter = {
        name     = "test2"
        required = false
        type     = "string"
      }
    }
  }
}
```

### Expected Behaviour

Query parameters named "test" and test2 should created for apimapio1 api_management_api_operation  resource.

### Actual Behaviour

The query parameters are not created and a terraform plan shows an empty request {} filed

### Steps to Reproduce

terraform apply

**Comment by jioblack** [comment]
This implementation don't look right https://github.com/aztfmod/terraform-azurerm-caf/blob/c82cfc65eecd0b8603c4e64effcfbda69c238bd5/modules/apim/api_management_api_operation/module.tf#L26

**Comment by arnaudlh** [comment]
Closing as merged in #1629
